resources/qml/delegates/ImageMessage.qml

```diff
@@ -31,11 +31,40 @@ Item {
         fillMode: Image.PreserveAspectFit
 
         MouseArea {
+            id: mouseArea
             enabled: model.data.type == MtxEvent.ImageMessage && img.status == Image.Ready
+            hoverEnabled: true
             anchors.fill: parent
             onClicked: TimelineManager.openImageOverlay(model.data.url, model.data.id)
         }
 
-    }
+        Item {
+            id: overlay
+        
+            anchors.fill: parent
+            visible: mouseArea.containsMouse
+            
+            Rectangle {
+                id: container
+                width: parent.width
+                implicitHeight: imgcaption.implicitHeight
```

Comment:
I think `height: imgcaption.height` would make more sense here.

Reply:
when I use `height: imgcaption.height` the text doesn't end up showing. I used the implicitHeight approach based on [this](https://stackoverflow.com/a/60929791/7138943). According to the QML docs for Item types, "Setting the implicit size is useful for defining components that have a preferred size based on their content" so I actually disagree with your statement, unless I'm interpreting it wrong?

Reply:
Yes, you are right, no idea why though :D